src/utils/utils.cpp

```diff
@@ -0,0 +1,15 @@
+#include "reffine/utils/utils.h"
+
+CUfunction compile_kernel(std::shared_ptr<Func> fn)
+{
+    reffine::CanonPass::Build(fn);
```

Comment:
why do you need reffine:: prefix? that namespace is already imported in the header file.